Consistent `--profile-name` option for setup commands

GeigerJ2 opened the pull request:
Currently, `verdi presto` uses the `--profile-name` option, while `verdi profile setup` uses `--profile`. I'd vouch for consistency.

GeigerJ2 commented:
Along these lines, I just realized that the options for the database hostname, port, username, and password for `verdi presto --use-postgres` are called:
`--postgres-hostname`
` --postgres-port`
` --postgres-username`
` --postgres-password`
(which makes sense as `--use-postgres` is explicitly specified, I guess).

While for `verdi profile setup core.psql_dos` they are called:
`--database-hostname`
`--database-port`
`--database-username`
`--database-password`

Any reason to keep them general in the second case (apart from backwards compatibility?). Are we planning to add another service-based database? Again, might be nice to have the options named consistently, so that people can switch seamlessly between writing a `verdi presto` and a `verdi profile setup` command (which is how I noticed now).

mbercx commented:
Checking the status of the AMD runners, will restart actions after.

unkcpz commented:
https://github.com/aiidateam/aiida-core/blob/022f049bfcf86609daf0c2d9ddc0b1c108b9ea7c/.docker/aiida-core-base/s6-assets/init/aiida-prepare.sh#L29

Should update this as well.

sphuber commented:
I am onboard with creating consistency, but is `--profile-name` not clearer? I would personally vote for that. We can add a short version `-p` to counter-act the increased length, which anyway would be a good idea.

GeigerJ2 commented:
> I am onboard with creating consistency, but is `--profile-name` not clearer? I would personally vote for that. We can add a short version `-p` to counter-act the increased length, which anyway would be a good idea.

Alright, that's fine for me, as long as there's also the short option ^^ I added another commit to revert the changes (rather than soft-resetting and force-pushing; to preserve the history), and reverted the files to their previous versions via `git checkout <SHA> -- </path/to/file.py>` in the first commit. Then, the second commit defines the new `SETUP_PROFILE_NAME` option. I didn't clone the previous `SETUP_PROFILE`, as that is only used in the deprecated `verdi setup` command, so can be removed once we remove that command, as well. I considered using this global `SETUP_PROFILE_NAME` in the `verdi presto` command, as well, rather than having it as a local `@click.option` as it is now, but that slightly changes the behavior, so I wouldn't do that. I assume there was a reason you added it as a local `@click.option`.

As such, the behavior of both commands stays the same, with `verdi profile setup` still being an `InteractiveOption`, while `verdi presto` still being a normal `@click.option`. The former now also uses `--profile-name`, and both allow `-p` as a shortcut.

Pinging @mbercx as the extra picky devil's advocate on CLI-related topics <3

GeigerJ2 commented:
> I am onboard with creating consistency, but is --profile-name not clearer? I would personally vote for that. We can add a short version -p to counter-act the increased length, which anyway would be a good idea.

Though, now that I'm changing the command in a bunch of places, I'm also thinking... this is backwards-incompatible with the previous version of `verdi profile setup` that uses `--profile`. It's a recent feature that was just introduced in v2.5, so it might be acceptable, but still something to consider? That being said, I'm also very much in favor of using `--profile-name` rather than `--profile` because `--profile` is also used when running `verdi` commands for a different profile than the default via `verdi --profile`. So good to have it distinctly different from that.

sphuber commented:
> Though, now that I'm changing the command in a bunch of places, I'm also thinking... this is backwards-incompatible with the previous version of verdi profile setup that uses --profile. It's a recent feature that was just introduced in v2.5, so it might be acceptable, but still something to consider? 

I think that is acceptable for now. Better to bite the bullet now while it hasn't really been adopted yet then having to struggle with it later on.

sphuber commented on src/aiida/cmdline/params/options/commands/setup.py:
```diff
@@ -181,6 +181,16 @@ def get_quicksetup_password(ctx, param, value):
 )
 
+SETUP_PROFILE_NAME = options.OverridableOption(
+    '-p',
+    '--profile-name',
```
```suggestion
    '-p',
    '--profile-name',
    'profile',
```
By adding this, I think the name of the argument that will be passed to the commands function definition will be `profile` instead of `profile_name`. This is preferable because the `ProfileParamType` will convert the profile name to a `Profile` instance. Now the `verdi profile setup` commands has a lot of `profile_name.name` which looks odd. Witht he suggested change it becomes `profile.name` which is better I think

GeigerJ2 replied:
Ah, that's a nice feature, and something I wondered about in the past, as well. I agree that it's nicer with using `profile` and added that in my recent commit.

sphuber commented on .molecule/default/setup_aiida.yml:
```diff
@@ -36,7 +36,7 @@
       {{ venv_bin }}/verdi setup
       --non-interactive
-      --profile "{{ aiida_backend }}"
+      --profile-name "{{ aiida_backend }}"
```
this is not correct is it? This is using `verdi setup` which is not being affected.

GeigerJ2 replied:
Good catch!

GeigerJ2 commented:
> I think that is acceptable for now. Better to bite the bullet now while it hasn't really been adopted yet then having to struggle with it later on.

I'm fine with that. @mbercx also mentioned we could deprecate `--profile` and hide it from being shown from `--help` (if that's possible), but not sure if it's even necessary (we can discuss quickly later). Apart from that, this should be ready for merge now.

sphuber commented:
Thanks @GeigerJ2
